packages/esm-admin-openconceptlab-app/src/subscription/subscription.component.tsx

```diff
@@ -92,14 +112,15 @@ const Subscription: React.FC = () => {
 
       return () => abortController.abort();
     },
-    [subscriptionUrl, token, validationType, isSubscribedToSnapshot, t, subscription, mutate],
+    [subscriptionUrl, token, validationType, isSubscribedToSnapshot, isSnapshotOptionDisabled, subscription, t, mutate],
   );
 
   const handleCancel = useCallback(() => {
     setSubscriptionUrl(subscription?.url || '');
     setToken(subscription?.token || '');
     setIsSubscribedToSnapshot(subscription?.subscribedToSnapshot || false);
     setValidationType(subscription?.validationType || 'FULL');
+    setIsSnapshotOptionDisabled(subscription ? isVersionDefinedInUrl(subscription.url) : false);
```

Comment:
If we're repeating this code multiple times in different places, it seems like maybe this should just be a `useEffect()` hook?

Reply:
This part is already included in the useEffect() hook in lines 34-40. Is there a way to trigger it when calling this `handleCancel` method?